src/ErrorBoundary.tsx

```diff
@@ -0,0 +1,18 @@
+import * as React from 'react';
+
+export interface ErrorBoundaryProps {}
+
+export class ErrorBoundary extends React.Component<ErrorBoundaryProps, any> {
+  state = { error: null };
+
+  componentDidCatch(error, errorInfo) {
+    this.setState({ error, errorInfo });
+  }
+
+  render() {
+    if (this.state.error) {
+      return null;
```

Comment:
We could be a bit creative here if desired, e.g. instead of returning null, we could return back the error that we caught and display it like so:

```
if (this.state.error) {
    return (<div>@ui-router/react-hybrid encountered an error: 
                       {JSON.stringify(this.state.error, null, 2)}
               </div>);
}
```

Reply:
But the error is lacking:

It just shows `Invariant violation`